core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -86,19 +85,11 @@ sealed trait Message[+F[_]] extends Media[F] { self =>
     val entity = w.toEntity(b)
 
     val hsBase = headers ++ w.headers
-    val cl: Option[`Content-Length`] = entity.length match {
-      case Some(l) =>
-        `Content-Length`
-          .fromLong(l)
-          .fold(
-            _ => {
-              Message.logger.warn(s"Attempt to provide a negative content length of $l")
-              None
-            },
-            Some(_),
-          )
-      case None => None
-    }
+    val cl: Option[`Content-Length`] = entity.length.flatMap(
+      `Content-Length`
+        .fromLong(_)
+        .toOption
+    )
```

Comment:
We can't log in this method unless we make `.withEntity` return a result in `F[_]` ... which seems like terrible ergonomics. So I just removed it.